tests/model/test_sam_resources.py

```diff
@@ -741,6 +741,119 @@ def test_function_datasource_set_with_none():
     assert none_datasource
 
 
+class TestSamFunctionRoleResolver(TestCase):
+    """
+    Tests for resolving IAM role property values in SamFunction
+    """
+
+    def setUp(self):
+        self.function = SamFunction("foo")
+        self.function.CodeUri = "s3://foobar/foo.zip"
+        self.function.Runtime = "foo"
+        self.function.Handler = "bar"
+        self.template = {"Conditions": {}}
```

Comment:
You don't need this `self.template` variable since it's not being used. Right below you use it to get `Conditions` from here, but it will be just `{}`, so you can just do that (`"conditions": {}`)